py/packages/genkit/src/genkit/blocks/prompt.py

```diff
@@ -679,3 +761,793 @@ async def render_user_prompt(
+    """Generate a registry definition key for a prompt.
+
+    Format: "ns/name.variant" where ns and variant are optional.
+
+    Args:
+        name: The prompt name.
+        variant: Optional variant name.
+        ns: Optional namespace.
+
+    Returns:
+        Registry key string.
+    """
+    parts = []
+    if ns:
+        parts.append(ns)
+    parts.append(name)
+    if variant:
+        parts[-1] = f"{parts[-1]}.{variant}"
+    return "/".join(parts)
+
+
+def registry_lookup_key(name: str, variant: str | None = None, ns: str | None = None) -> str:
+    """Generate a registry lookup key for a prompt.
+
+    Args:
+        name: The prompt name.
+        variant: Optional variant name.
+        ns: Optional namespace.
+
+    Returns:
+        Registry lookup key string.
+    """
+    return f"/prompt/{registry_definition_key(name, variant, ns)}"
+
+
+def define_partial(registry: Registry, name: str, source: str) -> None:
+    """Define a partial template in the registry.
+
+    Partials are reusable template fragments that can be included in other prompts.
+    In JS, files starting with `_` are treated as partials.
+
+    Args:
+        registry: The registry to register the partial in.
+        name: The name of the partial.
+        source: The template source code.
+    """
+    registry.dotprompt.define_partial(name, source)
+    logger.debug(f'Registered Dotprompt partial "{name}"')
+
+
+def define_helper(registry: Registry, name: str, fn: Callable) -> None:
+    """Define a Handlebars helper function in the registry.
+
+    Args:
+        registry: The registry to register the helper in.
+        name: The name of the helper function.
+        fn: The helper function to register.
+    """
+    registry.dotprompt.define_helper(name, fn)
+    logger.debug(f'Registered Dotprompt helper "{name}"')
+
+
+def load_prompt(
+    registry: Registry,
+    path: Path,
+    filename: str,
+    prefix: str = '',
+    ns: str = 'dotprompt'
+) -> None:
+    """Load a single prompt file and register it in the registry.
+
+    This function loads a .prompt file, parses it, and registers it as a lazy-loaded
+    prompt that will only be fully loaded when first accessed.
+
+    Args:
+        registry: The registry to register the prompt in.
+        path: Base path to the prompts directory.
+        filename: Name of the prompt file (e.g., "myPrompt.prompt" or "myPrompt.variant.prompt").
+        prefix: Subdirectory prefix (for namespacing).
+        ns: Namespace for the prompt.
+    """
+    # Extract name and variant from filename
+    # Matches JS behavior: prefix is included in name before variant extraction
+    # "myPrompt.prompt" -> name="myPrompt", variant=None
+    # "myPrompt.variant.prompt" -> name="myPrompt", variant="variant"
+    # "subdir/myPrompt.prompt" -> name="subdir/myPrompt", variant=None
+    base_name = filename[:-7]  # Remove ".prompt" extension
```

Comment:
look from the right for this substring and then remove it if required. 
this might be brittle if the input doesn't have `.prompt` in it.